File: datasets/IR-Plag/original-java-rest-kotlin/case-07/plagiarized/L6/01/L6.kt
package IR

import java.util.*

object L6 {
    fun jumlahDiagonal(m: Array<DoubleArray>): Double {
        var sum = 0.0
        var hit = m.size - 1

        while (hit >= 0) {
            sum += m[hit][hit]
            hit--
        }
        return sum
    }

    fun print(m: Array<DoubleArray?>?) {
        print("Sum of the elements in the major diagonal is " + jumlahDiagonal(m))
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val m = Array(4) { DoubleArray(4) }
        println("Enter a 4 by 4 matrix row by row: ")

        for (i in 3 downTo 0) {
            for (j in 3 downTo 0) {
                m[i][j] = sc.nextDouble()
            }
        }
        L6.print(m)
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L6/01/L6.py
import numpy as np

def jumlah_diagonal(m):
    sum = 0
    hit = m.shape[0] - 1
    
    while hit >= 0:
        sum += m[hit, hit]
        hit -= 1
    
    return sum

def print_result(m):
    print(f"Sum of the elements in the major diagonal is {jumlah_diagonal(m)}")

if __name__ == "__main__":
    m = np.array([list(map(float, input(f"Enter row {i+1} of the 4x4 matrix: ").split())) for i in range(4)])
    print_result(m)

